thoughts-of-a-rustacean-learning-go.md

```diff
@@ -0,0 +1,244 @@
+I've come to love the language and spend quite a bit of my free time contributing to the [compiler][compiler]
+and the [Servo browser engine][servo].
+
+// Perhaps a different form of introduction?
+
+
+[trpl]: http://rust-lang.org/
+[compiler]: https://github.com/rust-lang/rust
+[servo]: https://github.com/servo/servo
+
+Recently, for a course, we were told to use Go for the assignments and project. Whilst a tad disappointed
+that I couldn't use Rust for it, I saw this as a great opportunity. Rust and Go have been
+compared a lot as the "hot new languages", and finally I'd get to see the other side of the argument.
+
+By the end of the course, aside from some simpler assignments, I'd written
+[a basic implementation of the Raft consensus algorithm][raft] and a [toy application][dash] using it,
+both in Go.
+
+Before I get into the experience, let me preface this by mentioning that Rust and Go don't exactly target the same audiences.
+Go is garbage collected and is okay with losing out on some performance for ergonomics;
+whereas Rust tries to keep everything as a compile time check as much as possible.
+This usually makes Rust much more useful for lower level applications.
+
+In my specific situation, however, I was playing around with distributed systems via threads
+(or goroutines), so this fit perfectly into the area of applicability of both languages.
+
+
+This article isn't exactly intended to be a comparison between the two. I understand that as a newbie at Go,
+I'll be trying to do things the wrong way and make bad conclusions off of this. My way of coding may not be the
+"Go way" (I'm mostly carrying over my Rust style to my Go code since I don't know better); so everything may seem
+like a hack to me. Please keep this in mind whilst reading the post, and feel free to let me know the "Go way"
+of doing the things I was stumbling with.
+
+This is more of a sketch of my experiences with the language, specifically from the point of
+view of someone coming from Rust; used to the Rusty way of doing things. It might be useful as an
+advisory to Rustaceans thinking about trying the language out, and what to expect.
+
+[raft]: https://github.com/Manishearth/cs733/tree/master/assignment3/raft
+[dash]: http://manishearth.github.io/dash
+
+// I've structured this as "what I liked"/"what I didn't like", but I'm open to other ways of structuring this too
+
+## What I liked
+
+Despite the performance costs, having a GC at your disposal after using Rust for very long is quite liberating. 
+For a while my internalized borrow checker would throw red flags on me tossing around data indiscriminately,
+but I learned to  ignore it as far as Go code goes. I was able to quickly share state via pointers without
+worrying about safety, which was quite useful.
+
+Having channels as part of the language itself was also quite ergonomic.  The `data <- chan` and `chan <- data` syntax is fun to use,
```

Comment:
It would be great to have a side-by side comparison of the two styles of syntax here, rather than the inline `data <- chan` and `.send()`.